coops-ui/src/components/AnnualReport/Directors.vue

```diff
@@ -917,7 +911,7 @@ export default class Directors extends Mixins(DateMixin, ExternalMixin, EntityFi
         lastName: this.director.officer.lastName
       },
       deliveryAddress: { ...this.inProgressAddress },
-      mailingAddress: { ...this.inProgressMailAddress },
+      ...(!this.entityFilter(EntityTypes.Coop) && { mailingAddress: { ...this.inProgressMailAddress } }),
```

Comment:
Well this is a doozy.
Can it be refactored to something easier to understand, hopefully with if-then-else logic (or x ? y : z) ?
ATM, I can't guess what the result of the spread is when type isn't Coop.
Also I recommend the opposite logic for the if condition: "if BComp then" (otherwise you assume all other types except Coop have a mailing address). In this case I think it's more concise to specify for which types this applies.

Thanks. Sorry, this one broke my rule, "if it's hard to understand then it should be rewritten".

Reply:
I thought this was the cleanest solution to adding a conditional property but i can certainly try and refactor ( expand the code ) if that makes it simpler to read. 

Do we know which entity is the exception here? Either way, we are assuming one way or the other until we have business information about the other entities? Why don't we ask business about this!

Reply:
Yes please.

You're right, but today what we know for sure is that Mailing Address applies only to BComps so IMHO I would rather the code "opt in" for that case instead of opting out Coops. I could be wrong in the long term, but positive logic is always easier to understand :)

Reply:
Makes sense, i definitely agree. We can adjust it down the road if need be when we have other entities information.

Reply:
Ok, you changed the logic.